Firestore/Source/Public/FIRDocumentSnapshot.h

```diff
@@ -159,6 +160,18 @@ NS_SWIFT_NAME(DocumentSnapshot)
  */
 - (nullable id)objectForKeyedSubscript:(id)key;
 
+/**
+ * Retrieves a Timestamp field from the document.
+ *
+ * Using this method avoids converting the timestamp to NSDate, which might lose precision. If the
+ * field exists, but its type is not Timestamp, an exception will be thrown.
+ *
+ * @param key The field to retrieve.
+ *
+ * @return The timestamp contained in the field or `nil` if the field doesn't exist.
+ */
+- (nullable FIRTimestamp *)getTimestamp:(id)key;
```

Comment:
I'm surprised I didn't catch this looking at the API doc, but this is a highly non-idiomatic name as I commented above. The Objective-C-ish way of naming this would be `timestampForField`.

Note that `objectForKeyedSubscript` is an externally specified API that allows subscript notation to work against a document snapshot. We're using "key" there because that's what the API is documented as taking, but otherwise these values are "fields", not "keys".